Tools/PIDML/qaPidML.cxx

```diff
@@ -0,0 +1,367 @@
+      }
+
+      if (pdgCode == 2212) {
+        histReg.fill(HIST("contamination/2212in321"), pt);
+      }
+      if (pdgCode == -2212) {
+        histReg.fill(HIST("contamination/02212in321"), pt);
+      }
+
+      if (pdgCode == 321) {
+        histReg.fill(HIST("contamination/321in321"), pt);
+      }
+      if (pdgCode == -321) {
+        histReg.fill(HIST("contamination/0321in321"), pt);
+      }
+
+      if (pdgCode == 11) {
+        histReg.fill(HIST("contamination/11in321"), pt);
+      }
+      if (pdgCode == -11) {
+        histReg.fill(HIST("contamination/011in321"), pt);
+      }
+
+      if (pdgCode == 13) {
+        histReg.fill(HIST("contamination/13in321"), pt);
+      }
+      if (pdgCode == -13) {
+        histReg.fill(HIST("contamination/013in321"), pt);
+      }
+    }
+  }
+
+  template <typename T>
+  void fillMcHistos(const T& track, const int pdgCode)
+  {
+    // pions
+    if (pdgCode == 211) {
+      histReg.fill(HIST("MC/211"), track.pt());
+    } else if (pdgCode == -211) {
+      histReg.fill(HIST("MC/0211"), track.pt());
+    }
+    // protons
+    else if (pdgCode == 2212) {
+      histReg.fill(HIST("MC/2212"), track.pt());
+    } else if (pdgCode == -2212) {
+      histReg.fill(HIST("MC/02212"), track.pt());
+    }
+    // kaons
+    else if (pdgCode == 321) {
+      histReg.fill(HIST("MC/321"), track.pt());
+    } else if (pdgCode == -321) {
+      histReg.fill(HIST("MC/0321"), track.pt());
+    }
+    // electrons
+    else if (pdgCode == 11) {
+      histReg.fill(HIST("MC/11"), track.pt());
+    } else if (pdgCode == -11) {
+      histReg.fill(HIST("MC/011"), track.pt());
+    }
+    // muons
+    else if (pdgCode == 13) {
+      histReg.fill(HIST("MC/13"), track.pt());
+    } else if (pdgCode == -13) {
+      histReg.fill(HIST("MC/013"), track.pt());
+    } else {
+      histReg.fill(HIST("MC/else"), track.pt());
+    }
+  }
+
+  template <std::size_t i, typename T>
+  void fillPidHistos(const T& track, const int pdgCode, bool isPidTrue)
+  {
+    if (isPidTrue) {
+      histReg.fill(HIST(pidTrueRegistryNames[i]), track.pt());
+      histReg.fill(HIST(TPCPidTrueRegistryNames[i]), track.p(), track.tpcSignal());
+      histReg.fill(HIST("TPCSignalPidTrue"), track.p(), track.tpcSignal());
+      histReg.fill(HIST("TOFSignalPidTrue"), track.p(), track.beta());
+      histReg.fill(HIST(TOFPidTrueRegistryNames[i]), track.p(), track.beta());
+    } else {
+      histReg.fill(HIST(pidFalseRegistryNames[i]), track.pt());
+      histReg.fill(HIST(TPCPidFalseRegistryNames[i]), track.p(), track.tpcSignal());
+      histReg.fill(HIST("TPCSignalPidFalse"), track.p(), track.tpcSignal());
+      histReg.fill(HIST("TOFSignalPidFalse"), track.p(), track.beta());
+      histReg.fill(HIST(TOFPidFalseRegistryNames[i]), track.p(), track.beta());
+
+      double pt = track.pt();
+      fillContaminationRegistry(i, pdgCode, pt);
+    }
+  }
+
+  int pidParticle(const int pidLogits[])
```

Comment:
By the way, it would be less confusing if you renamed this to, e.g., `getParticlePdg()`.
`pidParticle` sounds like a class name.